Lib/SentryErrorHandler.php

```diff
@@ -1,47 +1,81 @@
 <?php
 
 
-/**
- * Description of SentryErrorHandler
- *
- * @author Sandreu
- */
-class SentryErrorHandler extends ErrorHandler {
-
-    protected static function sentryLog($exception) {
-        if (Configure::read('debug')==0 || !Configure::read('Sentry.production_only')) {
-            Raven_Autoloader::register();
-            App::uses('CakeRavenClient', 'Sentry.Lib');
-
-            $client = new CakeRavenClient(Configure::read('Sentry.PHP.server'));
-            $client->captureException($exception, get_class($exception), 'PHP');
-        }
-    }
+    /**
+     * Description of SentryErrorHandler
+     *
+     * @author Sandreu
+     */
+    class SentryErrorHandler extends ErrorHandler
+    {
+
+        protected static function sentryLog($exception)
+        {
+            if (Configure::read('debug') == 0 || !Configure::read('Sentry.production_only')) {
+                Raven_Autoloader::register();
+                App::uses('CakeRavenClient', 'Sentry.Lib');
 
-    public static function handleException($exception) {
-        try {
-            // Avoid bot scan errors
-            if (Configure::read('Sentry.avoid_bot_scan_errors') && ($exception instanceof MissingControllerException || $exception instanceof MissingPluginException) && Configure::read('debug')==0) {
-                echo Configure::read('Sentry.avoid_bot_scan_errors');
-                exit(0);
+                $client = new Raven_Client(Configure::read('Sentry.PHP.server'));
+                if (class_exists('AuthComponent')) {
+                    $model = Configure::read('Sentry.User.model');
+                    if (empty($model)) {
+                        $model = 'User';
+                    }
+                    $User = ClassRegistry::init($model);
+                    $mail = Configure::read('Sentry.User.email_field');
+                    if (empty($mail)) {
+                        if ($User->hasField('email')) {
+                            $mail = 'email';
+                        } else {
+                            if ($User->hasField('mail')) {
+                                $mail = 'mail';
+                            }
+                        }
+                    }
+                    $client->user_context(array(
+                        "is_authenticated" => AuthComponent::user($User->primaryKey) ? true : false,
+                        "id" => AuthComponent::user($User->primaryKey),
+                        "username" => AuthComponent::user($User->displayField),
+                        "email" => AuthComponent::user($mail)
+                    ));
+                }
+                $eventId = $client->captureException($exception, array(
+                    'extra' => array(
+                        'php_version' => phpversion(),
+                        'class' => get_class($exception)
+                    ),
+                ));
+
+                CakeSession::write('sentry_event_id',$eventId);
```

Comment:
This is a problem for 2 reasons:
- you might not be in a context where a session is established
- for the same reason, if you don't, you'll be greeted with this error: `Uncaught Error: Class 'CakeSession' not found in /…/app/Plugin/Sentry/Lib/SentryErrorHandler.php:49`

What was the reason to write this to the session here anyway?

This seems highly application specific to me and shouldn't be here in the first place. Or at least, be disabled by default.